### PR TITLE
Fix truncated body in exception view

### PR DESCRIPTION
**Pull request by timomeh**
ActionPack's `debug_exceptions` will calculate the Content-Length of the rendered body. [See implementation here.](https://github.com/rails/rails/blob/2929d165c23f0d3976425a8e70de77847cc4b872/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L180)

After injecting the console into the body, there will be a mismatch between the Content-Length Header and the actual response body. So the browser will truncate the output.

image

As a solution the Content-Length Header will be deleted, as the application server will calculate the Content-Length again, if it isn't present already.

### CODE REVIEW

File: lib/web_console/middleware.rb [hunk 1]
```diff
@@ -33,6 +33,11 @@ def call(env)
           headers["X-Web-Console-Session-Id"] = session.id
           headers["X-Web-Console-Mount-Point"] = mount_point
 
+          # Remove any previously set Content-Length header because we modify
+          # the body. Otherwise the response will be truncated.
+          # Someone will calculate it again, at least the application server.
+          headers.delete("Content-Length")
```
Comment (gsamokovarov) [inline lib/web_console/middleware.rb]: Can you add a test for this, so we don't regress again?

Comment (timomeh) [reply lib/web_console/middleware.rb]: I would be glad to do that, but unfortunately I don't have any clue how to test it. 😕

I tried adding a wrong Content-Length to the headers defined in `helper_test.rb` or `middleware_test.rb` to reproduce the bug in the first place, but even without this fix, the response in the test always contains the correct Content-Length. Maybe somewhere in the test-stack the Content-Length gets re-set to the correct value, where other application servers don't override the Content-Length if it's already set (e.g. puma)? My rails-knowledge isn't really that good...

Comment (gsamokovarov) [reply lib/web_console/middleware.rb]: Maybe we can explicitly set the `Content-Length` after the injection on line `42` ourselves? Then we can test that the content length is extended after the middleware run? That way, we won't depend on someone else to set the proper content length and therefore test our own behavior.

Comment (timomeh) [reply lib/web_console/middleware.rb]: The problem is that I cannot reproduce the bug (without my fix) in a test...

If I set `headers["Content-Length"] = 7` (just an arbitrary obviously wrong value) in the middleware instead of deleting it, the response in the test will always have the correct `Content-Length`, not the previously defined value of 7. It seems like something in the test library is setting `Content-Length` to the correct value all the time, which is not the same behavior as puma.

So if I add a test to check if `Content-Length` is set to the correct value, it will be green even without my fix... which kinda defeats the purpose of writing a test.

Personally, I don't think a library or application has to care about setting `Content-Length`, since this is usually the task of a server. That's why I would prefer deleting the header. I would go so far to say that the behavior of Action Pack is wrong, it shouldn't set the `Content-Length` in the first place.

So to summarize: If we delete `Content-Length` or re-calculate it in the middleware, I have no idea how to write a reasonable middleware test. 😅

Comment (timomeh) [reply lib/web_console/middleware.rb]: That being said, I only tried to test it using an Integration Test. I don't know how to properly test the middleware with another kind of test.

Comment (gsamokovarov) [reply lib/web_console/middleware.rb]: Maybe you can try to write a unit test (`ActiveSupport::TestCase`) with and execute the middleware manually with [rack-test](https://github.com/rack-test/rack-test). You can also move this behavior to the `WebConsole::Injector` and add another unit test to it, instead to the middleware. I'd be okay with the injector, caring about acting on the content length, even if we have to give it more input/output responsibilities.

Comment (timomeh) [reply lib/web_console/middleware.rb]: I just added a test without seeing your comment. 😅 But I added the middleware_test which always will be green because of the previously mentioned `IntegrationTest`.

Executing it manually sounds like a good plan. I'll add that (later this day, hopefully)

**Comment by gsamokovarov** [comment]
@timomeh, thanks for the change! I'll merge it as is and write a test in an upcoming commit. I have some time today and will like to release `3.6.2` with this fix soonish. ✌️

**Comment by timomeh** [comment]
@gsamokovarov Cool! I wrote a test yesterday using rack-test, but (who would have thought) it overrides the Content-Length Header, too. I'm looking forward to see how you tested it :)
